### DISCUSSION TITLE
Completely black screen when installing Kali M1 version

### DISCUSSION THREAD

**Discussion by bacchilega** [utmapp/UTM#4539] [UTM Help (macOS)] [answered]
When I try to install Kali Linux with graphical installation the screen go black.
Note: I'm using the M1 installer provided by kali linux website.

UTM Version:  3.2.4
Monterey 12.2.1
Mac M1 air 2020

**Comment by allytag** [comment]
I am also facing the same issue. :) UTM is not good...

**Comment by allytag** [comment]
Macbook is also shit...

**Comment by allytag** [comment]
when i am window usre i never face any issues

**Comment by bacchilega** [comment]
I like UTMm but if you really need to install Kali fast, try to install Parallels until this bug is not fixed.
Thank you UTM team.

**Comment by MichaelKofler** [comment]
Same problem. Was able to do a installation in text mode, graphics would run afterwards (i.e. after the installation has finished).

**Comment by MichaelKofler** [comment]
PS: also tried nomodeset kernel option, won't help. Neither video=1024x768.

**Comment by JCFlores93** [comment]
In my case after showing an error the screen become black.
Error:
UTM Version: 3.2.4
Monterey 12.4
Mac M1 pro 2020
```
mount: mounting /dev/vda on /media failed: Invalid argument
umount: can't umount /media: Invalid argument
mount: mounting /dev/vda on /media failed: Invalid argument
```

**Comment by ktprograms** [comment]
@JCFlores93 What is the OS running in the VM?

**Comment by JCFlores93** [comment]
This si Kali Linux

Obtener Outlook para Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by ktprograms** [comment]
At which point was this error shown? While booting, while running the installer, or something else?

**Comment by JCFlores93** [comment]
Yes, while running the installer. Exactly after I select the Graphic installer option.

**Comment by bacchilega** [comment]
I confirm. The error is encountered after select the graphic installer option

**Comment by adamwitherspoon-cars** [comment]
OS: Mac OS Monterey 12.4
MacBook Pro: M1 Pro (16-inch, 2021)
UTM: Version 3.2.4 (58)

Graphical installer fails with 
![Cursor_and_Screen_Recording_2022-06-16_at_4_54_03_PM_mov](https://user-images.githubusercontent.com/97473720/174196910-0e651446-bd06-4fa5-bf66-1f4abc36f7ce.png)
![Kali_and_UTM_and_Inbox__2_293__-_awitherspoon_creditiq_com_-_cars_com_Mail](https://user-images.githubusercontent.com/97473720/174196911-bd02e2e8-2042-4964-ac46-81d2e847bdfe.png)

then black screen and hangs

**Comment by nzmedet** [comment]
for anyone struggling, just install without graphics, its straightforward and easy. later it will give option to install desktop environment (checked by default)

After installation finished, move USB disk down in UTM settings, as it will try to install again.

**Reply by mmercurio** [reply]
Didn't work for me on M1 MacBook Pro running macOS 12.6.1. After install via serial console and rebooting, the virtual graphical display is still blank.

**Comment by freshprince96** [comment]
Im having the same issue. After I select "graphical Install" I get the same error message, followed by a completely dark screen. Anyone ever solved this issue? Does UTM know of this issue? Im currently running Mac OS 12.4 M1 chip.

**Comment by phenric** [comment]
> for anyone struggling, just install without graphics, its straightforward and easy. later it will give option to install desktop environment (checked by default)
> 
> After installation finished, move USB disk down in UTM settings, as it will try to install again.

I did what @vko-online suggested and it works like a charm.

**Comment by deFr0ggy** [comment]
It seems like there is an issue with UTM, I checked the same and it seems like Kali Linux is not loading either with Graphics Mode and with no graphics. 

The screen gets stuck for both.

**Comment by ghost** [comment]
Running into the exact same problem on an M1 mac running 12.5. Both graphical and text installs hang indefinitely.

This is [where](https://i.imgur.com/ppFq6gg.png) my install hangs.

**Comment by bacchilega** [comment]
@rmangino same issue with both

**Comment by jhonsay** [comment]
Hi to everyBody. I was the same problem like you and I can fix it with the following steps.

1º- Edit the VM in UTM and **change in Display the TYPE from Full Graphics to Text only**.
2º- Then run the VM with the ISO of Kali Linux and select the first kind of installation. In the progress you can install a GUI interface.
3º- When the installation was finished, log in and shutdown. 
4º- Edit the VM again and change in Display the TYPE to Full Graphics.
5º- Done.

The problem is the Installation program, it hasn't got a correct display drivers.

Luck !

**Comment by deFr0ggy** [comment]
I tried the same before though the issue remained the same. I have moved onto Parallels for Mac and it works like a charm. Better than VirtualBox and VMWare TBH!

**Comment by jhonsay** [comment]
did you try exactly what i said?, step by step?. It's work perfect and with free open source platform :)

**Comment by phiratio** [comment]
> Hi to everyBody. I was the same problem like you and I can fix it with the following steps.
> 
> 1º- Edit the VM in UTM and **change in Display the TYPE from Full Graphics to Text only**. 2º- Then run the VM with the ISO of Kali Linux and select the first kind of installation. In the progress you can install a GUI interface. 3º- When the installation was finished, log in and shutdown. 4º- Edit the VM again and change in Display the TYPE to Full Graphics. 5º- Done.
> 
> The problem is the Installation program, it hasn't got a correct display drivers.
> 
> Luck !

This is working 
thanks!

**Comment by frederikblais** [comment]
> Hi to everyBody. I was the same problem like you and I can fix it with the following steps.
> 
> 1º- Edit the VM in UTM and **change in Display the TYPE from Full Graphics to Text only**. 2º- Then run the VM with the ISO of Kali Linux and select the first kind of installation. In the progress you can install a GUI interface. 3º- When the installation was finished, log in and shutdown. 4º- Edit the VM again and change in Display the TYPE to Full Graphics. 5º- Done.
> 
> The problem is the Installation program, it hasn't got a correct display drivers.
> 
> Luck !

This worked for me thanks!

**Comment by ghost** [comment]
Thanks you so much @jhonsay this was exactly my issue!! I could carry on with the installation without any problem :)

**Comment by jhonsay** [comment]
> Thanks you so much @jhonsay this was exactly my issue!! I could carry on with the installation without any problem :)

Happy to read you :)

**Comment by MhTab-ai** [comment]
Hi, I run the steps as per @jhonsay but still facing an issue. 
Interactive shell display the following. How to "select the first kind of installation"?

Thank you.  

****
UEFI Interactive Shell v2.2                                                                                
EDK II                                                                                                     
UEFI v2.70 (EDK II, 0x00010000)                                                                            
Mapping table                                                                                              
     BLK1: Alias(s):                                                                                       
          VenHw(93E34C7E-B50E-11DF-9123-2543DFD72085,00)                                                   
     BLK0: Alias(s):                                                                                       
          PciRoot(0x0)/Pci(0x4,0x0)                                                                        
Press ESC in 1 seconds to skip startup.nsh or any other key to continue.                                   
Shell>                                                                                                     
****

**Comment by ghost** [comment]
If that's showing up, maybe you need to put the ISO image before start the machine. When the VM is selected (but not launched), there is some information like Status, Architecture, Machine... In CD/DVD the ISO must be selected, then carry on and launch the VM. After installation, you can remove the ISO

**Comment by GibzB** [comment]
> Hi to everyBody. I was the same problem like you and I can fix it with the following steps.
> 
> 1º- Edit the VM in UTM and **change in Display the TYPE from Full Graphics to Text only**. 2º- Then run the VM with the ISO of Kali Linux and select the first kind of installation. In the progress you can install a GUI interface. 3º- When the installation was finished, log in and shutdown. 4º- Edit the VM again and change in Display the TYPE to Full Graphics. 5º- Done.
> 
> The problem is the Installation program, it hasn't got a correct display drivers.
> 
> Luck !

Worked the charm for me! Thank you!

**Comment by Tr0x69** [comment]
For anyone is still having problem installing linux or having back screen error. Follow this video https://www.youtube.com/watch?v=Z2e-fRLxXnU, but before setting up the machine, go to linux setting and go to display and change the 'type' to console only. Then follow to the video. When ur done with setting up the machine, change back to Full graphic and reboot the machine. It worked for me.

**Comment by arturradiuk** [comment]
> Hi to everyBody. I was the same problem like you and I can fix it with the following steps.
> 
> 1º- Edit the VM in UTM and **change in Display the TYPE from Full Graphics to Text only**. 2º- Then run the VM with the ISO of Kali Linux and select the first kind of installation. In the progress you can install a GUI interface. 3º- When the installation was finished, log in and shutdown. 4º- Edit the VM again and change in Display the TYPE to Full Graphics. 5º- Done.
> 
> The problem is the Installation program, it hasn't got a correct display drivers.
> 
> Luck !

Worked for me, thanks

**Comment by jaswal-shubham** [comment]
@jhonsay thank you so much bro it worked like charm for me. was facing this issue from last one week. thank you so much

**Comment by barvhaim** [comment]
> Hi to everyBody. I was the same problem like you and I can fix it with the following steps.
> 
> 1º- Edit the VM in UTM and **change in Display the TYPE from Full Graphics to Text only**. 2º- Then run the VM with the ISO of Kali Linux and select the first kind of installation. In the progress you can install a GUI interface. 3º- When the installation was finished, log in and shutdown. 4º- Edit the VM again and change in Display the TYPE to Full Graphics. 5º- Done.
> 
> The problem is the Installation program, it hasn't got a correct display drivers.
> 
> Luck !

worked!

**Comment by MhTab-ai** [comment]
Hi, I tried the entire download and configuration as described by @barvhaim, but when all is set, Kali does not reboot and asks to re-do the configuration again. any help please?

**Comment by ogbozoyan** [comment]
> Hi to everyBody. I was the same problem like you and I can fix it with the following steps.
> 
> 1º- Edit the VM in UTM and **change in Display the TYPE from Full Graphics to Text only**. 2º- Then run the VM with the ISO of Kali Linux and select the first kind of installation. In the progress you can install a GUI interface. 3º- When the installation was finished, log in and shutdown. 4º- Edit the VM again and change in Display the TYPE to Full Graphics. 5º- Done.
> 
> The problem is the Installation program, it hasn't got a correct display drivers.
> 
> Luck !

bruuuuuh where you was before

**Comment by MhTab-ai** [comment]
<img width="876" alt="Capture d’écran 2022-08-29 à 08 36 24" src="https://user-images.githubusercont
<img width="876" alt="Capture d’écran 2022-08-29 à 08 36 52" src="https://user-images.githubusercontent.com/111655865/187138721-1f8e6648-12a9-4948-92ff-1ef3f7e00121.png">
ent.com/111655865/187138717-50708cb2-8bfe-42af-ad10-9aadfdfd0c1e.png">

**Comment by ogbozoyan** [comment]
> <img width="876" alt="Capture d’écran 2022-08-29 à 08 36 24" src="https://user-images.githubusercont
> 
> <img width="876" alt="Capture d’écran 2022-08-29 à 08 36 52" src="https://user-images.githubusercontent.com/111655865/187138721-1f8e6648-12a9-4948-92ff-1ef3f7e00121.png">
> 
> ent.com/111655865/187138717-50708cb2-8bfe-42af-ad10-9aadfdfd0c1e.png">
> 
> 

After installation you should clear cd/dvd and switch back display mode

**Comment by MhTab-ai** [comment]
<img width="876" alt="Capture d’écran 2022-08-29 à 08 36 24" src="https://user-images.githubusercontent.com/111655865/187139051-ee2ebfc4-ea98-4097-97ce-a1be4716873c.png">

**Comment by ogbozoyan** [comment]
> <img width="876" alt="Capture d’écran 2022-08-29 à 08 36 24" src="https://user-images.githubusercontent.com/111655865/187139051-ee2ebfc4-ea98-4097-97ce-a1be4716873c.png">
> 
> 

Show utm

**Comment by MhTab-ai** [comment]
I completed the installation but when I press continue, I get the window to redo it again ?

**Comment by ogbozoyan** [comment]
> I completed the installation but when I press continue, I get the window to redo it again ?

After complete you should clear cd/dvd in utm

**Comment by MhTab-ai** [comment]
<img width="913" alt="Capture d’écran 2022-08-29 à 08 43 42" src="https://user-images.githubusercontent.com/111655865/187139390-e1b067a8-1e07-43ed-aa19-71c2182c3d50.png">

**Comment by ogbozoyan** [comment]
> <img width="913" alt="Capture d’écran 2022-08-29 à 08 43 42" src="https://user-images.githubusercontent.com/111655865/187139390-e1b067a8-1e07-43ed-aa19-71c2182c3d50.png">
> 
> 
Click cd/dvd -> clear

**Comment by MhTab-ai** [comment]
yes done. How to close kali now 
<img width="913" alt="Capture d’écran 2022-08-29 à 08 45 53" src="https://user-images.githubusercontent.com/111655865/187139748-725b8aae-f62a-4b29-a9f4-76ad1dcbf057.png">
?

**Comment by ogbozoyan** [comment]
> yes done. How to close kali now 
> 
> <img width="913" alt="Capture d’écran 2022-08-29 à 08 45 53" src="https://user-images.githubusercontent.com/111655865/187139748-725b8aae-f62a-4b29-a9f4-76ad1dcbf057.png">
> 
> ?

Comman + q

**Comment by MhTab-ai** [comment]
Many many thanks It is working

**Comment by ogbozoyan** [comment]
> Many many thanks It is working

wich mac you got ?

**Comment by MhTab-ai** [comment]
MacBook Air , M1, 2020

**Comment by ogbozoyan** [comment]
> MacBook Air , M1, 2020
> 
> 
> On Mon, Aug 29, 2022 at 10:43 AM ogizoyani_:) <
> ***@***.***> wrote:
> 
> > Many many thanks It is working
> >
> > wich mac you got ?
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <https://github.com/utmapp/UTM/issues/4090#issuecomment-1229963252>, or
> > unsubscribe
> > <https://github.com/notifications/unsubscribe-auth/A2T3XONTRUYNCHWKBYRDFATV3RZ3HANCNFSM5XWVSPHQ>
> > .
> > You are receiving this because you commented.Message ID:
> > ***@***.***>
> >
> 

everything works smoothly? 
I've few days ago got m1 pro 14 inch base model, I thought that it will "fly", but in real life it was not so smoothly

**Comment by TheIlluminatiS** [comment]
> Hi to everyBody. I was the same problem like you and I can fix it with the following steps.
> 
> 1º- Edit the VM in UTM and **change in Display the TYPE from Full Graphics to Text only**. 2º- Then run the VM with the ISO of Kali Linux and select the first kind of installation. In the progress you can install a GUI interface. 3º- When the installation was finished, log in and shutdown. 4º- Edit the VM again and change in Display the TYPE to Full Graphics. 5º- Done.
> 
> The problem is the Installation program, it hasn't got a correct display drivers.
> 
> Luck !

Thanks, I try this and works perfectly fine.

**Comment by ghost** [comment]
> Hi to everyBody. I was the same problem like you and I can fix it with the following steps.
> 
> 1º- Edit the VM in UTM and **change in Display the TYPE from Full Graphics to Text only**. 2º- Then run the VM with the ISO of Kali Linux and select the first kind of installation. In the progress you can install a GUI interface. 3º- When the installation was finished, log in and shutdown. 4º- Edit the VM again and change in Display the TYPE to Full Graphics. 5º- Done.
> 
> The problem is the Installation program, it hasn't got a correct display drivers.
> 
> Luck !

This worked for me. Thank you so much.

**Comment by Homy** [comment]
> Hi to everyBody. I was the same problem like you and I can fix it with the following steps.
> 
> 1º- Edit the VM in UTM and **change in Display the TYPE from Full Graphics to Text only**. 2º- Then run the VM with the ISO of Kali Linux and select the first kind of installation. In the progress you can install a GUI interface. 3º- When the installation was finished, log in and shutdown. 4º- Edit the VM again and change in Display the TYPE to Full Graphics. 5º- Done.

Дякую!

**Comment by lukbukkit** [comment]
With the latest apt updates for Kali Linux, a functioning Kali Linux installation with a working GUI just turns black on reboot. I can still interact with it via the shell, but the screen is just black (except the cursor). Does anyone have a clue why?

I'm using UTM 3.2.4 and Kali Linux 2022.3 with the latest version of all available pre-installed packages.

**Comment by diegorv** [comment]
> With the latest apt updates for Kali Linux, a functioning Kali Linux installation with a working GUI just turns black on reboot. I can still interact with it via the shell, but the screen is just black (except the cursor). Does anyone have a clue why?
> 
> I'm using UTM 3.2.4 and Kali Linux 2022.3 with the latest version of all available pre-installed packages.

Easy to fix

![CleanShot 2022-09-28 at 19 53 44](https://user-images.githubusercontent.com/24455/192902570-6bcf5da4-f05c-44d8-9e1b-cd7a336b7481.png)
![CleanShot 2022-09-28 at 19 54 02](https://user-images.githubusercontent.com/24455/192902601-0e4e325e-8696-4664-a4cb-b0202ecaccca.png)
![CleanShot 2022-09-28 at 19 58 06](https://user-images.githubusercontent.com/24455/192903022-e4249306-f893-4231-be4a-0765b166db6f.png)

Change `Emulated Display card` from `virtio-ramfb-gl` to `virtio-ramfb` or `virtio-gpu-pci`.

Looks like a display driver problem 😢 

You can read this [link](https://www.kraxel.org/blog/2019/09/display-devices-in-qemu/) to understand the differences of _Display Devices in QEMU_

**Reply by phallikainen** [reply]
This was it. For me only virtio-ramfb worked - any "gl" won't. Running M1 version with Ventura 13.0.1.

**Reply by Tobils** [reply]
save my day by changing emulator diplay card

Thanks buddy

**Reply by mmercurio** [reply]
Confirmed this worked for me on M1 MacBook Pro using the latest Kali weekly build (2023-W02) with UTM 4.1.5. Thank you! 🎉

**Reply by floodoo** [reply]
This works for me (m2 mac) Thank You!

**Comment by optikfluffel** [comment]
> 1º- Edit the VM in UTM and change in Display the TYPE from Full Graphics to Text only.

I'm running UTM version `4.0.8` and the mentioned "Full Graphics" and "Text only" options are gone. 😞

**Reply by AurelVU** [reply]
Now to enable only text mode, do the following:
1) Go to the virtual machine settings
2) Remove display
3) Click New in Devices and add serial

**Reply by Mamasodikov** [reply]
It's working solution, thanks!

**Reply by anqiwoo** [reply]
Thank you!

**Comment by DarkShadowG** [comment]
> > 1º- Edit the VM in UTM and change in Display the TYPE from Full Graphics to Text only.
> 
> I'm running UTM version `4.0.8` and the mentioned "Full Graphics" and "Text only" options are gone. 😞

same its not working can anyone help?

**Comment by optikfluffel** [comment, marked as answer]
@DarkShadowG here you go: https://docs.getutm.app/guides/kali/#black-screen-on-start

**Reply by juanmercadoit** [reply]
this is a solution. I try it and is clear.
1. create a new driver.
2. select SERIAL
3. start the installation. Using the text mode in the ISO menu options, when starting the installation.
4. when finished with the installation, only remove the SERIAL device. 

<img width="803" alt="Screenshot 2022-11-17 at 13 00 24" src="https://user-images.githubusercontent.com/61954396/202495889-19b68a4b-7e36-476c-a84a-eab47cf9fdf9.png">

I put here the evidence of my case and test in some M1 more.
<img width="1290" alt="Screenshot 2022-11-17 at 13 04 25" src="https://user-images.githubusercontent.com/61954396/202496792-ae14733e-e67f-43e3-9bfc-44c219e5178c.png">

I hope that you have a funny installation!
Enjoy Kali ;)

**Reply by Lukas-io** [reply]
THANK YOU SO MUCH! THIS SHOULD BE PINNED SOMEWHERE.

**Reply by RootDev4** [reply]
This solution worked for me (Apple M3 Pro), thank you very much!
During installation, two windows are displayed. One for the actual Kali installation (with the flashing cursor) and the other with the terminal session (serial). After installation, close the serial window and delete the serial device from menu. 

<img width="810" height="530" alt="Bildschirmfoto 2026-02-04 um 12 30 48" src="https://github.com/user-attachments/assets/3aa6c222-9737-4ab2-b638-1d30b97f3b1b" />

**Comment by yassirrah** [comment]
@DarkShadowG same here , cannot change display type to Text only with this new version of UTM

**Comment by yassirrah** [comment]
@optikfluffel issue solved , thanks man !

**Comment by constantinos91-beep** [comment]
Anyone find a solution? I have the same issue black screen after clicking on instal

**Reply by osy** [reply]
Did you try the selected answer above?

**Comment by basilbenaziz** [comment]
If you are using Kali 2022.2 use the older version 2022.1 this fixed it for me

This worked and got me going in a pinch before I had read the kali docs.
A full list of kali releases can be found [here](https://old.kali.org/kali-images/).
Select the [2022.1 release](https://old.kali.org/kali-images/kali-2021.1/)
There's three types of arm64 installers. You probably want default installer.
If you need a different installer the differences are explained [here](https://www.kali.org/docs/introduction/what-image-to-download/).

**Reply by avishj** [reply]
Just as a FYI, you linked to 2021.1 instead of 2022.1.

Also, this works: https://docs.getutm.app/guides/kali/

**Comment by muhajirinfatona** [comment]
i got black screen while installing kali linux in UTM. can anyone help??

i am using UTM Version 4.0.8

**Reply by avishj** [reply]
Follow this, https://docs.getutm.app/guides/kali/

**Reply by juanmercadoit** [reply]
I use the same version that Avish and I followed the steps in the link, and I solved the issue.
Above I shared the same solution with photos.
Regards, 

Juan Andres

**Comment by masteralb** [comment]
Spent some time with this issue,

Seems that the issue is not related to UTM itself, because if you try to install kali-linux-2022.2-installer-arm64.iso everything works fine.

So the issue is happening only with kali-linux-2022.3-installer-amd64.iso

Hope this helps

**Comment by LaurEncord** [comment]
Hey I just hit this issue too and overcame it by adding a Serial within UTM and using that to navigate the install where the main GUI wasn't coming up. See screenshots. Worked with `kali-linux-2022.4-installer-arm64.iso` Hope this helps!

![Screenshot 2022-12-24 at 19 50 11](https://user-images.githubusercontent.com/119512483/209449463-69978fcd-bfb9-4494-8fa5-2f31b48606dd.png)
![Screenshot 2022-12-24 at 19 50 35](https://user-images.githubusercontent.com/119512483/209449464-bd4ca9dc-48fd-487b-b25b-5806b1b9e72c.png)

**Comment by hvaandres** [comment]
The solution to this problem is the following:

1.- You will have to choose: The display option | Emulated Display Card - virtio-ramfb
2.- Add a Serial under Devices from the settings options of the VM
3.- You will get a screen asking you what type of software you want to install. Then, you will have to install Kali Linux without any software, you will have to uncheck all of the software that has been checked by default. 
4.- After you get a prompt window with "The installation is ready" You have to eject the ISO image
5.- Restart the machine, and login with your credentials
6.- Run the following commands:

```
sudo apt update
sudo apt install -y kali-desktop-kde
sudo update-alternatives --config x-session-manager
```

<img width="783" alt="Screenshot 2023-01-14 at 8 47 47 PM" src="https://user-images.githubusercontent.com/26829975/212521980-c7507d26-1eef-4253-9867-d3bcc12ef650.png">

<img width="886" alt="Screenshot 2023-01-14 at 8 50 56 PM" src="https://user-images.githubusercontent.com/26829975/212522032-9be29439-e7af-4f33-a98d-3064232217f3.png">

<img width="1093" alt="Screenshot 2023-01-14 at 8 56 04 PM" src="https://user-images.githubusercontent.com/26829975/212522154-30e5065d-5ea7-4553-818c-5524f4a6d0e1.png">


This will solve the black screen. I just downloaded the latest version of Kali and UTM, please ping me if you have any questions. If you would like to watch a video about this, you can check out this link [How to install kali Linux through UTM](https://www.youtube.com/watch?v=C-Aa8_iB5Xs) OR [check out my blog about this issue](https://ithvaandres.medium.com/black-screen-while-installing-kali-linux-on-macs-m1-m2-through-utm-5d5939d93577)

**Reply by rwielinga** [reply]
Thanks, my issue was not so much installing kali (using the serial device / terminal as per instructions worked well).
It was after the installation that the graphical screen remained black.

Setting the emulated display card to virtio-ramfb instead of virtio-ramfb-gl (GPU Supported) did the trick.

**Reply by RichardoC** [reply]
Changing the virtual display card to virtio-ramfb also worked for me. Thank you!
![image](https://user-images.githubusercontent.com/11242848/218781031-d66786f2-094d-4725-8003-816fe0cf9ebf.png)

**Comment by Shankar775** [comment]
Please tell me how to fix black screen on nethunter kex app

**Comment by NiekSulter** [comment]
For people who have enabled LVM with disk encryption: when the disk is encrypted the display will remain blank, I had to input the passphrase via the serial console. After which, the display started working again.

kali-linux-2024.1-arm64
UTM 4.5.2

**Comment by mbahgojol** [comment]
I have the same issue. it happened when I did `sudo apt get upgrade`
if you are also like me, follow this step to fix this issue:

- add serial devices
- run your kali-linux
- put `fsck -y /dev/vda2` in your terminal serial (it's will repair your system)
- and reboot

And this is my result after that.
<img width="592" alt="image" src="https://github.com/utmapp/UTM/assets/19856599/8b763016-4858-4aa0-a4bb-2bb54c576794">

**Comment by mengbo** [comment]
Change to Wayland, everything is OK.
```
more /usr/lib/systemd/system/gdm.service.d/disable-wayland.conf
```
